### ISSUE TITLE
Reconstruction 

### ISSUE THREAD

**Issue by Lucasmogsan** [eriksandstroem/Loopy-SLAM#7]
![Selection_001](https://github.com/user-attachments/assets/f283fda2-5f5d-4418-bd44-b7c833d3f24c)

Hi,
First of all thanks for the great work done on Loopy- and Point-SLAM!
I have a question on the reconstruction quality:
As I understand...

- The images saved in mapping_vis or tracking_vis (left on the image attached) are the reconstruction during training (just after optimizing to a specific pose and camera input). Is that correct?
- The rendered_image is (right in the image attached) is the reconstruction looping through the poses after the full scene is optimized and stored in the MLP?

I guess the obvious question is why this degrades so much and to me the obvious answer is that it is due to the MLP having to contain the whole scene for the rendered_image whereas the mapping_vis is just after the optimization to a specific image? How wrong am I? ;-)

Thanks.

**Comment by Lucasmogsan** [comment]
@eriksandstroem is this interpretation correct about the intermediate (left) and final (right) render?

**Comment by eriksandstroem** [comment]
Hi @Lucasmogsan ,
Sorry for my late reply. 
On your first questions, you are correct on both - the rendering accuracy online is better than the final offline one because when you render online, you optimize perfectly for that view but for the final renderings, you need a representation that can render from any view and therefore it degrades a bit. Further reasons for this effect are: when we rendering online we only render from the single active submap, but when we render at the end, we render from the merged global map.

So in conclusion, your guess is on track for sure - since we optimize the color MLP all the time, it is actually only accurate in the online setting and at the end, we use a suboptimal MLP for each submap (only the last submap should have a perfect MLP). We did some studies on the final refinement in the paper and came to the conclusion that this was very important, due to the catastrophic forgetting of the color MLP otherwise.

Hope this helps! And thank you for appreciating our work! That means a lot!

**Comment by Lucasmogsan** [comment]
Thanks a lot for your answer, that explains it perfectly.
